fineract-core/src/main/java/org/apache/fineract/commands/service/SynchronousCommandProcessingService.java

```diff
@@ -201,6 +203,14 @@ private void setIdempotencyKeyStoreFlag(boolean flag) {
         fineractRequestContextHolder.setAttribute(IDEMPOTENCY_KEY_STORE_FLAG, flag);
     }
 
+    @Override
+    public void logFailedBatchRequestWithEnclosingTransaction(final CommandWrapper commandRequest, final BatchResponse failedBatchResult) {
+        final AppUser user = context.authenticatedUser(commandRequest);
+        final CommandSource commandSource = CommandSource.fullEntryForBatchFailed(commandRequest, failedBatchResult, user,
+                idempotencyKeyGenerator.create());
```

Comment:
I do not think that generating a new idempotency key is correct, rather we should read it from the incoming request header and save that. The main purpose of the idempotency is that the caller knows it and if it tries to reuse (from same or different instance) then the same result (without perform anything) is returned.